pkg/controller/volume/expand/expand_controller.go

```diff
@@ -211,10 +211,12 @@ func (expc *expandController) pvcUpdate(oldObj, newObj interface{}) {
 		volumeSpec := volume.NewSpecFromPersistentVolume(pv, false)
 		volumePlugin, err := expc.volumePluginMgr.FindExpandablePluginBySpec(volumeSpec)
 		if err != nil || volumePlugin == nil {
-			err = fmt.Errorf("didn't find a plugin capable of expanding the volume; " +
-				"waiting for an external controller to process this PVC")
-			eventType := v1.EventTypeNormal
-			if err != nil {
+			var eventType string
+			if err == nil {
```

Comment:
instead of moving these blocks into if/else statements you could just take the old code and just move the `err = fmt.errorf...` block to after the `if err != nil` block